redis/src/main/scala/zio/redis/executor/RedisConnection.scala

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2021 John A. De Goes and the ZIO contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.redis.executor
+
+import zio._
+import zio.redis.{RedisError, RedisUri, logScopeFinalizer}
+import zio.stream.{Stream, ZStream}
+
+import java.io.{EOFException, IOException}
+import java.net.{InetSocketAddress, SocketAddress, StandardSocketOptions}
+import java.nio.ByteBuffer
+import java.nio.channels.{AsynchronousSocketChannel, Channel, CompletionHandler}
+
+private[redis] trait RedisConnection {
```

Comment:
If I am not mistaken, this was formerly known as `ByteStream`? While I think that the name change is good, there are two problems associated with it:
- package hierarchy restructuring
- rename diff

I would ask you to refrain from making non-related changes and restore the previous structure (you can keep the new name).

Reply:
I think it's good time to restructure package hierarchy and rename classes other way with cluster class it would be full mess. It's better to do now in first steps. And think a little bit further about where structure would develop.

Reply:
What is messy? Top-level package has ~10 artifacts. Please don't pollute a very worthy pull request with package reorganization that is not necessary.

Reply:
Ok. If i revert this structure in current PR, can I open another PR with this reorganization of folders?

Reply:
I am against it in general case. We don't need to split a package with ~10 artifacts into 5 packages with 2. It just makes navigation more difficult, and doesn't bring any value. If it grows significantly, then I don't object doing it.

Reply:
Ok. I guess we found main disagreement) It's ok to have disagreements about code structure.))
If you find this structure is a bad idea of course it should be reverted and would be done when we find agreement)

In my opinion it is opposite. A proper structure help navigate through project and make shape of it, that new people can easily find that they want and understand where to put stuff. 

Reverted